### PR TITLE
nfs: add github action for NFS integration test

### LINKED ISSUE #7934: Convert NFS integration tests to run in a github action
**Is this a bug report or feature request?**
* Feature Request

**What should the feature do:**
The NFS integration tests should be run by a github action. For example, the NFS tests can follow the same pattern as the Cassandra tests that were converted in #7601.

**What is use case behind this feature:**
The goal is to run all tests in github actions and turn off Jenkins CI soon. The NFS tests are one of the remaining gaps in this conversion.

### PR DESCRIPTION
**Pull request by subhamkrai**
**Description of your changes:**

this commit add github action for NFS
integration test.

Signed-off-by: subhamkrai <srai@redhat.com>

**Which issue is resolved by this Pull Request:**
Resolves #7934 

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

**Comment by subhamkrai** [comment]
@travisn do we want to remove stage https://github.com/rook/rook/blob/cfcdcac56ebb31b6af2191b7009ab9485b1789f3/Jenkinsfile#L120 from Jenkins 

or just skip like here https://github.com/rook/rook/blob/cfcdcac56ebb31b6af2191b7009ab9485b1789f3/Jenkinsfile#L212

File: .github/workflows/integration-test-nfs-suite.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+name: Integration test NfsSuite
```
Comment (leseb) [inline .github/workflows/integration-test-nfs-suite.yaml:1]:
```suggestion
name: Integration test NFS Suite
```

File: .github/workflows/integration-test-nfs-suite.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+on:
+  pull_request:
+    branches:
+      - master
+      - release-*
+jobs:
+  TestNfsSuite:
```
Comment (leseb) [inline .github/workflows/integration-test-nfs-suite.yaml:8]:
```suggestion
  TestNFSSuite:
```

**Review by leseb — CHANGES_REQUESTED** [review]

File: .github/workflows/integration-tests-on-release.yaml [hunk 3]
```diff
@@ -305,3 +305,54 @@ jobs:
...
+      run: |
+        # set VERSION to a dummy value since Jenkins normally sets it for us. Do this to make Helm happy and not fail with "Error: Invalid Semantic Version"
+        GOPATH=$(go env GOPATH) make clean && make -j$nproc IMAGES='nfs' VERSION=0 build
+        docker images
+        docker tag $(docker images|awk '/build-/ {print $1}') rook/nfs:master
+
+    - name: TestNfsSuite
+      run: |
+        export DEVICE_FILTER=$(lsblk|awk '/14G/ {print $1}'| head -1)
+        SKIP_CLEANUP_POLICY=false go test -v -timeout 1800s -run NfsSuite github.com/rook/rook/tests/integration
```
Comment (travisn) [inline .github/workflows/integration-tests-on-release.yaml:346]: I don't think we need `DEVICE_FILTER` or `SKIP_CLEANUP_POLICY` for the nfs test
```suggestion
        go test -v -timeout 1800s -run NfsSuite github.com/rook/rook/tests/integration
```

File: .github/workflows/integration-tests-on-release.yaml [hunk 4]
```diff
@@ -305,3 +305,54 @@ jobs:
...
+    - name: build rook
+      run: |
+        # set VERSION to a dummy value since Jenkins normally sets it for us. Do this to make Helm happy and not fail with "Error: Invalid Semantic Version"
+        GOPATH=$(go env GOPATH) make clean && make -j$nproc IMAGES='nfs' VERSION=0 build
+        docker images
+        docker tag $(docker images|awk '/build-/ {print $1}') rook/nfs:master
+
+    - name: TestNfsSuite
+      run: |
+        export DEVICE_FILTER=$(lsblk|awk '/14G/ {print $1}'| head -1)
```
Comment (travisn) [inline .github/workflows/integration-tests-on-release.yaml:345]:
```suggestion
```

**Review by travisn — CHANGES_REQUESTED** [review]
> @travisn do we want to remove stage
> 
> https://github.com/rook/rook/blob/cfcdcac56ebb31b6af2191b7009ab9485b1789f3/Jenkinsfile#L120
> 
> from Jenkins
> or just skip like here
> 
> https://github.com/rook/rook/blob/cfcdcac56ebb31b6af2191b7009ab9485b1789f3/Jenkinsfile#L212

Let's not worry about removing NFS from Jenkins in this PR. In a future PR we can just remove the integration tests stage as your first suggestion.

File: .github/workflows/integration-test-nfs-suite.yaml (hunk 1 above)
Comment (subhamkrai) [reply .github/workflows/integration-test-nfs-suite.yaml]: done

File: .github/workflows/integration-tests-on-release.yaml (hunk 3 above)
Comment (subhamkrai) [reply .github/workflows/integration-tests-on-release.yaml]: removed.

**Comment by subhamkrai** [comment]
@rohan47 installing nfs-common worked. Thanks

**Review by travisn — APPROVED** [review]

**Pull request merged**